draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,607 @@
+
+## `_deleg` label presence {#presence}
+
+Absence of the `_deleg` label in a zone is a clear signal that the zone does not contain any incremental deleg delegations.
+Recursive resolvers SHOULD NOT send any additional incremental deleg queries for zones for which it is known that it does not contain the `_deleg` label at the apex.
+The state regarding the presence of the `_deleg` label within a resolver can be "unknown", "known not to be present", or "known to be present".
+
+When the presence of a `_deleg` label is "unknown", a `_deleg` presence test query SHOULD be send in parallel to the next query for the zone.
+The query name for the test query is the `_deleg` label prepended to the apex of zone for which to test presence, with query type A.
+
+The testing query can have three possible outcomes:
+
+1. The `_deleg` label does not exist within the zone, and an NXDOMAIN response is returned.
+
+   The non-existence of the `_deleg` label MUST be cached for the duration indicated by the "minimum" RDATA field of the SOA resource record in the authority section, adjusted to the boundaries for TTL values that the resolver has ({{Section 4 of !RFC8767}}).
+   For the period the non-existence of the `_deleg` label is cached, the label is "known not to be present" and the resolver SHOULD NOT send any (additional) incremental deleg queries.
+
+2. The `_deleg` label does exist within the zone but contains no data.
+   A NOERROR response is returned with no RRs in the answer section.
+
+   The existence of the `_deleg` name MUST be cached for the duration indicated by the "minimum" RDATA field of the SOA resource record in the authority section, adjusted to the resolver's TTL boundaries.
+   For the period the existence of the empty non-terminal at the `_deleg` label is cached, the label is "known to be present" and the resolver MUST send additional incremental deleg queries as described in TODO.
+
+3. The `_deleg` label does exist within the zone and contains data.
+   A NOERROR response is return with an A RRset in the answer section.
+
+   The returned A RRset in the answer section MUST be cached for the duration indicated by the TTL for the RRset, adjusted to the resolver's TTL boundaries.
+   For the period any RRset at the `_deleg` label is cached, the label is "known to be present" and the resolver MUST send additional incremental deleg queries as described in TODO.
+
+## Authoritative name server support
+
+Incremental delegations supporting authoritative name servers do not need the additional (parallel) incremental delegation query, but instead will include the incremental delegation information (or the NSEC(3) records showing the non-existence) in the authority section of referral responses.
+If it is known that an authoritative name server supports incremental deleg, than no direct queries for the incremental delegation need to be send in parallel to the legacy delegation query.
```

Comment:
/s/than/then/  and /s/send/sent/    (sentence is in passive voice, thus its sent and not send)